### PR TITLE
Feature/venue backend - Event Table connected

### PR DESCRIPTION
**Pull request by ihysi2024**
Connected event scroll bar to backend

### CODE REVIEW

File: backend/internal/models/event.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Represents Venue Event
+type Event struct {
+	EventID   int       `json:"event_id"`
+	Name      string    `json:"name"`
+	Date      time.Time `json:"event_date"`
```
Comment (garrettladley) [inline backend/internal/models/event.go:13]: having an `Event.Date` of type `time.Time` and an `Event.Time` of type `time.Time` is redundant

File: backend/internal/models/event.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	Time      time.Time `json:"event_time"`
+	ImagePath string    `json:"image_path"`
+	VenueID   uuid.UUID `json:"venue_id"`
+	CreatedAt time.Time `json:"created_at"`
+}
+
+func (e *Event) Validate() map[string]string {
+	return nil
```
Comment (garrettladley) [inline backend/internal/models/event.go:21]: - [ ] fill out this stub. probably want simple checks on string len of [name, image path], etc etc based on requirements

File: backend/internal/utils/validators.go [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+package utils
+
+import (
+	"regexp"
+)
+
+// Check if the string is a valid email address
+func IsEmail(input string) bool {
+	// Define the regular expression for validating an email address
+	var emailRegex = regexp.MustCompile(`^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`)
```
Comment (garrettladley) [inline backend/internal/utils/validators.go:11]: these regex are being compiled on each function call. they should be precompiled in a private var, then used in this function

File: backend/internal/utils/validators.go [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+	return emailRegex.MatchString(input)
+}
+
+// Check if the string is a valid UUID
+func IsUUID(input string) bool {
```
Comment (garrettladley) [inline backend/internal/utils/validators.go:16]: use `uuid.Parse` instead.

```go
var raw string
// populate raw somehow
uuid, err := uuid.Parse(raw)
if err != nil {
// !valid, handle me
}
```

File: Makefile [hunk 5]
```diff
@@ -1,3 +1,77 @@
+<<<<<<< HEAD
```
Comment (garrettladley) [inline Makefile:1]: finish resolving this merge conflict?

File: Makefile [hunk 6]
```diff
@@ -1,3 +1,77 @@
...
+.PHONY: backend-build
+backend-build:
+	cd backend && go build -o bin/nightlife cmd/server/main.go
+
+# convert the backend link to an ngrok link
+.PHONY: backend-ngrok
+backend-ngrok:
+	@echo ${EXPO_PUBLIC_API_DOMAIN}
+	cd backend && ngrok http --domain=${EXPO_PUBLIC_API_DOMAIN} 8080
+=======
```
Comment (garrettladley) [inline Makefile:74]: finish resolving this merge conflict?

File: Makefile [hunk 7]
```diff
@@ -69,4 +143,5 @@ backend-build:
 .PHONY: backend-ngrok
 backend-ngrok:
 	@echo ${EXPO_PUBLIC_API_DOMAIN}
-	cd backend && ngrok http --url=${EXPO_PUBLIC_API_DOMAIN} 8080
\ No newline at end of file
+	cd backend && ngrok http --url=${EXPO_PUBLIC_API_DOMAIN} 8080
+>>>>>>> b6136ca773a9f06b8b6704803b38115da574bbaa
```
Comment (garrettladley) [inline Makefile:147]: finish resolving this merge conflict?

**Review by garrettladley — CHANGES_REQUESTED** [review]

**Review by CamPlume1 — APPROVED** [review]

**Comment by CamPlume1** [comment]
Closing after checking in with Ira

**Pull request closed without merging**